packages/core/src/react-integration/provider/BackupBoundary.tsx

```diff
@@ -1,15 +1,26 @@
 import React, { Suspense, memo, useMemo, version } from 'react';
 
 /* istanbul ignore next  */
-const NoSuspense =
-  (version.startsWith('16') || version.startsWith('17')) &&
-  typeof window === 'undefined';
+const LegacyReact = version.startsWith('16') || version.startsWith('17');
+/* istanbul ignore next  */
+const SSR = typeof document === 'undefined';
+const SSR_ID = 'rest-hooks-SSR';
```

Comment:
what happens if there's more than one render tree on the page?

Reply:
and only one does SSR and the other does not? well the only differences is a div inserted with a specific ID so while this would 'change behavior' otherwise I don't see it actually breaking anything as a non-SSR thing couldn't care less exactly what is rendered.

In other words - we must make sure SSR and client match; in this worst case scenario we simply lose some debugging capability but there are no errors.

Reply:
The whole point of the backup boundary is for people who forget to put suspense somewhere